.gitlab-ci.yml

```diff
@@ -40,46 +68,123 @@ dev:
     - rake destroy
     - rake clobber
 
-promote-to-stg:
+gcp-dev:
+  stage: dev
+  tags:
+  - gcp
+  script:
+    - cd gcp/live/dev
+    - rake clobber
+    - rake configure_serviceaccount_ci
+    - rake
+  after_script:
+    # Clean up even if something failed.
+    - cd gcp/live/dev
+    - rake destroy
+    - rake clobber
+
+aws-promote-to-stg:
+  stage: promote-to-stg
+  tags:
+  - aws
+  script:
+    - export DATESTAMP="$(date -u '+%Y%m%d%H%M%S')"  # e.g. 20170603220542. Uses UTC.
+    - echo "DATESTAMP is $DATESTAMP"  # So it's visible in the log
+    - git tag "deploy-aws-stg-$DATESTAMP"
+    # gitlab is not clever enough to clean up an added remote and git complains
+    # if we add a remote that already exists.
+    - git remote | grep -q "^origin-rw" || git remote add origin-rw git@github.com:gpii-ops/gpii-infra
```

Comment:
although this works, other option could be setting the origin url using:
`git remote set-url origin git@github.com:gpii-ops/gpii-infra`
it's just an idea.

Reply:
I couldn't find it in my notes, but I think I considered this idea when I wrote this task originally. I decided it was safer to create a separate origin rather than fight for control of `origin` with Gitlab, which may have certain expectations about the origin URL that we might break.

I checked and the origin url Gitlab uses is `https://gitlab-ci-token:<PASSWORD>@gitlab.com/gpii-ops/gpii-infra.git`. So I feel like there might be consequences if we change this to SSH-based authentication. :)

I propose we keep this as-is.